Orange/widgets/visualize/owscatterplot.py

```diff
@@ -619,5 +620,38 @@ def test_main(argv=None):
 
     return rval
 
+
+class Test(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
```

Comment:
Only use setUpClass when it makes sense.

setUp method is called before each test is run for a reason. Try removing the last two tests (*_none) and run all remaining tests. Now the test_subset_data fails, as self.data is still set to iris.